test/src/TimeoutExtender.spec.js

```diff
@@ -0,0 +1,137 @@
+  })
+  it('tracks multiple messages', () => {
+    inst = new TimeoutExtender(new SquissStub())
+    inst.addMessage(fooMsg)
+    inst.addMessage(barMsg)
+    inst._index.should.have.property('foo')
+    inst._index.should.have.property('bar')
+  })
+  it('deletes a head node', () => {
+    inst = new TimeoutExtender(new SquissStub())
+    inst.addMessage(fooMsg)
+    inst.addMessage(barMsg)
+    inst.addMessage(bazMsg)
+    inst.deleteMessage(fooMsg)
+    inst._head.message.raw.MessageId.should.equal('bar')
+  })
+  it('deletes a tail node', () => {
+    inst = new TimeoutExtender(new SquissStub())
+    inst.addMessage(fooMsg)
+    inst.addMessage(barMsg)
+    inst.addMessage(bazMsg)
+    inst.deleteMessage(bazMsg)
+    inst._tail.message.raw.MessageId.should.equal('bar')
+  })
+  it('deletes a middle node', () => {
+    inst = new TimeoutExtender(new SquissStub())
+    inst.addMessage(fooMsg)
+    inst.addMessage(barMsg)
+    inst.addMessage(bazMsg)
+    inst.deleteMessage(barMsg)
+    inst._head.next.message.raw.MessageId.should.equal('baz')
+  })
+  it('renews a message approaching expiry', () => {
+    clock = sinon.useFakeTimers(100000)
+    const squiss = new SquissStub()
+    const spy = sinon.spy(squiss, 'changeMessageVisibility')
+    inst = new TimeoutExtender(squiss, { visibilityTimeoutSecs: 10 })
+    inst.addMessage(fooMsg)
+    spy.should.not.be.called()
+    clock.tick(6000)
+    spy.should.be.calledOnce()
+  })
+  it('emits "timeoutExtended" on renewal', done => {
+    clock = sinon.useFakeTimers(100000)
+    const squiss = new SquissStub()
+    squiss.on('timeoutExtended', msg => {
+      msg.should.equal(fooMsg)
+      done()
+    })
+    inst = new TimeoutExtender(squiss, { visibilityTimeoutSecs: 10 })
+    inst.addMessage(fooMsg)
+    clock.tick(6000)
+  })
+  it('renews two messages approaching expiry', () => {
+    clock = sinon.useFakeTimers(100000)
+    const squiss = new SquissStub()
+    const spy = sinon.spy(squiss, 'changeMessageVisibility')
+    inst = new TimeoutExtender(squiss, { visibilityTimeoutSecs: 20 })
+    inst.addMessage(fooMsg)
+    clock.tick(10000)
+    inst.addMessage(barMsg)
+    spy.should.not.be.called()
+    clock.tick(10000)
+    spy.should.be.calledOnce()
+    clock.tick(10000)
+    spy.should.be.calledTwice()
+    clock.tick(10000)
+    spy.should.be.calledThrice()
+  })
+  it('renews only until the configured age limit', () => {
+    clock = sinon.useFakeTimers(100000)
+    const squiss = new SquissStub()
+    const spy = sinon.spy(squiss, 'changeMessageVisibility')
+    inst = new TimeoutExtender(squiss, { visibilityTimeoutSecs: 10, noExtensionsAfterSecs: 15 })
+    inst.addMessage(fooMsg)
+    clock.tick(10000)
+    spy.should.be.calledOnce()
+    clock.tick(20000)
+    spy.should.be.calledOnce()
+  })
+  it('emits error on the parent Squiss object in case of issue', done => {
+    clock = sinon.useFakeTimers(100000)
+    const squiss = new SquissStub()
+    squiss.on('error', () => done())
```

Comment:
So it's been a while since I've used `done()` but I'm guessing this test would fail -- with some sort of timeout maybe?! -- if done never gets called?